Cython/Tests/TestShadow.py

```diff
@@ -77,6 +77,17 @@ def test_int_types_in_shadow(self):
                         missing_types.append(ptr_name)
         self.assertEqual(missing_types, [])
 
-    # TODO - there's a lot of types that are looked up by `cython_scope.lookup_type` that
-    # it's unfortunately hard to get a definite list of to confirm that they're present
-    # (because they're obtained by on-the-fly string parsing)
+    def test_most_types(self):
+        # TODO it's unfortunately hard to get a definite list of types to confirm that they're
+        # present (because they're obtained by on-the-fly string parsing in `cython_scope.lookup_type`)
+        missing_types = []
+        for (signed, longness, name), type_ in PyrexTypes.modifiers_and_name_to_type.items():
+            if name in ['ssize_t', 'ptrdiff_t', 'object']:
```

Comment:
I'm not sure whether/if `ssize_t` and `ptrdiff_t` should be excluded here. I suspect they should probably be in Shadow.py instead.

Reply:
`ssize_t` is POSIX and not standard C, that's why there is `Py_ssize_t` which is always defined by Python. It's safer to use that instead.

`ptrdiff_t` is C99 and seems fine to add.

Reply:
I'm fairly sure `from cython cimport ssize_t` would work (although maybe I should test that too...) although it'd presumably only compile on POSIX, so on that basis user's might expect to find it in Shadow.py?

Reply:
Right. It seems bad to allow it in compiled code but not in `Shadow.py`. OTOH, `ssize_t` is just the normal signed `size_t`, so maybe we should exclude `usize_t` instead of adding an explicit `ssize_t` and excluding lots of things?